hack/.ci/deploy-release-gke.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2024 ScyllaDB
+#
+
+set -euExo pipefail
+shopt -s inherit_errexit
+shopt -s extglob
+
+if [ -z "${ARTIFACTS+x}" ]; then
+  echo "ARTIFACTS can't be empty" > /dev/stderr
+  exit 2
+fi
+
+source "$( dirname "${BASH_SOURCE[0]}" )/lib/e2e.sh"
+parent_dir="$( dirname "${BASH_SOURCE[0]}" )"
+
+trap gather-artifacts-on-exit EXIT
+
+SO_NODECONFIG_PATH="${SO_NODECONFIG_PATH=${parent_dir}/manifests/cluster/nodeconfig.yaml}"
+export SO_NODECONFIG_PATH
+
+SCYLLA_OPERATOR_FEATURE_GATES="${SCYLLA_OPERATOR_FEATURE_GATES:-AllAlpha=true,AllBeta=true}"
+export SCYLLA_OPERATOR_FEATURE_GATES
```

Comment:
Contrary to `/hack/ci-deploy.sh`, it's not being used in `/hack/ci-deploy-release.sh` script. And it's also already being set in `/hack/.ci/run-e2e-gke-release.sh`. It also doesn't feel platform-dependent?

Reply:
removed

Reply:
When this script gets wired into `/hack/.ci/run-e2e-gke-release.sh`, wouldn't it make sense to move all of [these env vars](https://github.com/scylladb/scylla-operator/blob/master/hack/.ci/run-e2e-gke-release.sh#L20-L27)  into this script and actually wire `SCYLLA_OPERATOR_FEATURE_GATES` to be used in `/hack/ci-deploy-release.sh`? Doesn't have to be in this PR but it seems like it would unify the usage better

Reply:
not against it eventually, but I can also see how the machines/nodes, or disks can differ

Reply:
> not against it eventually, but I can also see how the machines/nodes, or disks can differ

both of those scripts target the same platform though? Moving it to this script would just cover the setup better with `e2e-gke-release-script-latest`, no?

Reply:
say the regular GKE one will have big machines with many cores or fancy/multiple NVME - you don't need to spin such a beast to verify you can run a deploy-release script

Reply:
ok, fair